### PR TITLE
Feature/event attendee logic

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

File: app/src/main/java/com/artemissoftware/tasky/agenda/data/repositories/EventRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        val syncType = if (event.syncState == SyncType.SYNCED) SyncType.UPDATE else event.syncState
+        eventDao.upsertSyncStateAndEvent(
+            eventEntity = event.toEntity(),
+            attendees = event.attendees.map { it.toEntity(eventId = event.id) },
+        )
+
+        return try {
+            // TODO: complete when remote part is ready. Next PR
+
+            DataResponse.Success(Unit)
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/data/repositories/EventRepositoryImpl.kt:30]: Here you'd change the syncState back to SYNCED, right?

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailState.kt [hunk 2]
```diff
@@ -23,6 +22,7 @@ data class EventDetailState(
     val notification: NotificationType = NotificationType.defaultNotification(),
     val attendeeDialogState: AttendeeDialogState = AttendeeDialogState(),
     val attendees: List<Attendee> = emptyList(),
+    val host: String = "",
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailState.kt:25]: If this is an id, i'd rename it to "hostId"

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt [hunk 3]
```diff
@@ -259,9 +286,22 @@ class EventDetailViewModel @Inject constructor(
...
+                result?.let { item ->
+                    _state.update {
+                        it.copy(
+                            agendaItem = item,
+                            notification = NotificationType.getNotification(remindAt = item.remindAt, startDate = item.from),
+                            startDate = item.from,
+                            title = item.title,
+                            description = item.description ?: "",
+                            endDate = item.to,
+                            attendees = item.attendees,
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt:301]: What about the host id? You should also add it from the db to the state

**Review by MKiperszmid — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/agenda/data/repositories/EventRepositoryImpl.kt (hunk 1 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/data/repositories/EventRepositoryImpl.kt:30]: yes :)

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailState.kt (hunk 2 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailState.kt:25]: Refactored

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt (hunk 3 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt:301]: Refactored

**Pull request merged**